src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,273 @@
+package duke.parser;
+
+import duke.commands.AddCommand;
+import duke.commands.ByeCommand;
+import duke.commands.Command;
+import duke.commands.DeleteCommand;
+import duke.commands.DoneCommand;
+import duke.commands.FindCommand;
+import duke.commands.ListCommand;
+import duke.commands.TasksOnCommand;
+import duke.exceptions.EmptyDescriptionException;
+import duke.exceptions.IncorrectFormatException;
+import duke.exceptions.InvalidKeywordException;
+import duke.exceptions.InvalidNumberInputException;
+import duke.exceptions.NonExistentTaskException;
+import duke.tasklist.TaskList;
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.ToDo;
+import duke.utils.Constants;
+
+import java.time.LocalDate;
+
+/**
+ * The Parser class encapsulates the parsing, interpretation and validation of the user's input commands
+ *
+ */
+public class Parser {
+
+    /**
+     * The parse method interprets the keyword of the user's input and accordingly calls the other task-specific
+     * parse functions
+     *
+     * @param input user's string input
+     * @param taskList the list of tasks to be operated upon
+     * @return a command returned by the respective task-specific parse function
+     */
+    public static Command parse(String input, TaskList taskList) {
+        String[] parsedInput = input.trim().split("\\s", 2);
+        String command = parsedInput[0];
+
+        switch(command.toUpperCase()) {
+            case "BYE":
```

Comment:
There should be no indentation for case clauses according to Java coding standard.